### PR TITLE
Exoplanets should no longer repeat

### PR DESCRIPTION
**Pull request by Alberyk**
This pr makes sure that two exoplanets of the same type don't spawn. This should make things more diverse.

### CODE REVIEW

**Review by SleepyGemmy — APPROVED** [review]

File: maps/_common/mapsystem/map.dm [hunk 1]
```diff
@@ -166,6 +166,7 @@
 		var/exoplanet_type = pick(possible_exoplanets)
 		log_debug("Building new exoplanet with type: [exoplanet_type] and size: [planet_size[1]] [planet_size[2]]")
 		var/obj/effect/overmap/visitable/sector/exoplanet/new_planet = new exoplanet_type(null, planet_size[1], planet_size[2])
+		possible_exoplanets -= new_planet
```
Comment (Sindorman) [inline maps/_common/mapsystem/map.dm:169]: Are you sure that will remove it from the list? `possible_exoplanets` is a type based list, `new_planet` is a reference to an object of the type. I think you need to use `new_planet.type`

**Review by Sindorman — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]
double check what Sindorman commented.

**Review by SleepyGemmy — APPROVED** [review]

File: maps/_common/mapsystem/map.dm (hunk 1 above)
Comment (Alberyk) [reply maps/_common/mapsystem/map.dm]: It runtimes now because the list is empty somehow.

**Review by NonQueueingMatt — APPROVED** [review]

**Comment by SleepyGemmy** [comment]
```
runtime error:
[03:44:40]pick() from empty list
[03:44:40]proc name: build exoplanets (/datum/map/proc/build_exoplanets)
[03:44:40]  source file: map.dm,166
[03:44:40]  usr: null
[03:44:40]  src: Runtime Ship (/datum/map/runtime)
[03:44:40]  call stack:
[03:44:40]Runtime Ship (/datum/map/runtime): build exoplanets()
[03:44:40]Mapping (/datum/controller/subsystem/mapping): Initialize(134658)
[03:44:40]Mapping (/datum/controller/subsystem/mapping): StartInitialize(134658)
[03:44:40]Master (/datum/controller/master): Initialize(10, 0)
```
DNM until fixed.

File: maps/_common/mapsystem/map.dm [hunk 2]
```diff
@@ -162,11 +162,12 @@
...
-		var/exoplanet_type = pick(possible_exoplanets)
-		log_debug("Building new exoplanet with type: [exoplanet_type] and size: [planet_size[1]] [planet_size[2]]")
-		var/obj/effect/overmap/visitable/sector/exoplanet/new_planet = new exoplanet_type(null, planet_size[1], planet_size[2])
-		new_planet.build_level()
+	if(possible_exoplanets.len)
+		for(var/i = 0, i < num_exoplanets, i++)
+			var/exoplanet_type = pick_n_take(possible_exoplanets)
+			log_debug("Building new exoplanet with type: [exoplanet_type] and size: [planet_size[1]] [planet_size[2]]")
+			var/obj/effect/overmap/visitable/sector/exoplanet/new_planet = new exoplanet_type(null, planet_size[1], planet_size[2])
+			new_planet.build_level()
```
Comment (Sindorman) [inline maps/_common/mapsystem/map.dm:170]: Here is a better way
```suggestion
var/exoplanets_to_spawn = min(possible_exoplanets.len, num_exoplanets)
for(var/i = 0, i < exoplanets_to_spawn, i++)
	var/exoplanet_type = pick_n_take(possible_exoplanets)
	log_debug("Building new exoplanet with type: [exoplanet_type] and size: [planet_size[1]] [planet_size[2]]")
	var/obj/effect/overmap/visitable/sector/exoplanet/new_planet = new exoplanet_type(null, planet_size[1], planet_size[2])
	new_planet.build_level()
```

**Review by SleepyGemmy — APPROVED** [review]

**Review by Sindorman — APPROVED** [review]

**Pull request merged**